x/stakeibc/keeper/reward_converter.go

```diff
@@ -0,0 +1,442 @@
+
+	// Send the ICA tx to kick off transfer from hostZone through rewardZone to the tradeZone (no callbacks)
+	tradeZoneConnectionId := route.TradeToHostHop.FromAccount.ConnectionId
+	err := k.SubmitICATxWithoutCallback(ctx, tradeZoneConnectionId, types.ICAAccountType_TRADE, msgs, timeout)
+	if err != nil {
+		return errorsmod.Wrapf(types.ErrICATxFailed, "Failed to submit ICA tx, Messages: %+v, err: %s", msgs, err.Error())
+	}
+
+	return nil
+}
+
+// Trade reward tokens in the Trade ICA for the target output token type using ICA remote tx on trade zone
+// The amount represents the total amount of the reward token in the trade ICA found by the calling ICQ
+// Depending on min and max swap amounts set in the route, it is possible not the full amount given will swap
+func (k Keeper) TradeRewardTokens(ctx sdk.Context, amount sdk.Int, route types.TradeRoute) error {
+	// If the min swap amount was not set it would be ZeroInt, if positive we need to compare to the amount given
+	//  then if the min swap amount is greater than the current amount, do nothing this epoch to avoid small swaps
+	if route.MinSwapAmount.IsPositive() && route.MinSwapAmount.GT(amount) {
+		return nil
+	}
+
+	// If the max swap amount was not set it would be ZeroInt, if positive we need to compare to the amount given
+	//  then if max swap amount is LTE to amount full swap is possible so amount is fine, otherwise set amount to max
+	if route.MaxSwapAmount.IsPositive() && route.MaxSwapAmount.GT(amount) {
+		amount = route.MaxSwapAmount
+	}
+
+	// See if pool swap spot price has been set to a valid ratio (string representing a float like "10.203")
+	// If there is a valid spot price, use it to set a floor for the acceptable minimum output tokens
+	// 5% slippage is allowed so multiply by 0.95 the expected spot price to get the target minimum
+	// minOut is the minimum number of route.TargetDenomOnTradeZone we must receive or the swap will fail
+	minOut := sdk.ZeroInt()
+	if route.SpotPrice != "" {
+		if spotPrice, err := sdk.NewDecFromStr(route.SpotPrice); err == nil {
+			slippageRatio := sdk.NewDecWithPrec(95, 2) // 0.95 to allow 5% loss to slippage
+			inputToOutputRatio := slippageRatio.Mul(spotPrice)
+			minOut = sdk.NewDecFromInt(amount).Mul(inputToOutputRatio).TruncateInt()
+		} else {
+			k.Logger(ctx).Error("Couldn't parse the spot price %s as a Decimal", route.SpotPrice)
+			// Don't allow a missing spot price to stop the swap, use zeroInt as the lower bound for now
+		}
+	}
+
+	tradeIcaAccount := route.RewardToTradeHop.ToAccount
+	tradeTokens := sdk.NewCoin(route.RewardDenomOnTradeZone, amount)
+
+	// Prepare Osmosis GAMM module MsgSwapExactAmountIn from the trade account to perform the trade
+	// If we want to generalize in the future, write swap message generation funcs for each DEX type,
+	// decide which msg generation function to call based on check of which tradeZone was passed in
+	var msgs []proto.Message
+	if amount.GT(sdk.ZeroInt()) {
```

Comment:
I'm assuming this check was in here as a mistake cause we verify above that amount is greater than 0 and, if it was zero, we'd want to exit instead of continuing with sending an empty ICA

Reply:
Yes, it was a stand in for the min_swap_amount before that was defined on the traderoute.  Now that we do those checks before reaching this, I think we can remove it no issue.